presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkRunner.java

```diff
@@ -63,26 +63,39 @@ public PrestoSparkRunner(PrestoSparkDistribution distribution)
     }
 
     public void run(
+            String user,
+            Optional<Principal> principal,
```

Comment:
Curious about the usage of `principal`, `extraCredentials` and `traceToken`? -- probably relate to security story?

Reply:
Yeah. I just wanted to expose all required parameters since I'm already modifying the interface.